tests/cli/test_cli_core.py

```diff
@@ -173,46 +221,67 @@ def test_delete_nonexisting(cli_runner, trailblazer_context, caplog):
         assert "Analysis not found" in caplog.text
 
 
-def test_delete_ongoing_fail(cli_runner, trailblazer_context, caplog):
-    with caplog.at_level("ERROR"):
-        # GIVEN Trailblazer database with analyses and jobs
-        trailblazer_context["trailblazer"].update_ongoing_analyses()
+def test_delete_ongoing_fail(
+    cli_runner: CliRunner,
+    trailblazer_context: Dict[str, MockStore],
+    caplog,
+    ongoing_analysis_case_name: str,
+    process_exit_success: int,
+):
+    """Test deleting an ongoing analysis without using the force flag."""
+    caplog.set_level("ERROR")
+
+    # GIVEN Trailblazer database with analyses and jobs
+    trailblazer_db: MockStore = trailblazer_context["trailblazer"]
+    trailblazer_db.update_ongoing_analyses()
 
-        # GIVEN an analysis that is ongoing
-        analysis_obj = trailblazer_context["trailblazer"].get_latest_analysis(case_id="escapedgoat")
+    # GIVEN an analysis that is ongoing
+    analysis: Optional[Analysis] = trailblazer_db.get_latest_analysis_for_case(
+        case_name=ongoing_analysis_case_name
+    )
 
-        # WHEN trying to delete ongoing analysis without --force flag
-        result = cli_runner.invoke(delete, [str(analysis_obj.id)], obj=trailblazer_context)
+    # WHEN trying to delete ongoing analysis without --force flag
+    result = cli_runner.invoke(delete, [str(analysis.id)], obj=trailblazer_context)
 
-        # THEN command executes successfully
-        assert result.exit_code == 0
+    # THEN command executes successfully
+    assert result.exit_code == process_exit_success
```

Comment:
Feels a bit weird to me that we consider this a success. Probably not within the scope of this PR even if it should be changed, though.

Reply:
Agreed -we should raise an error, but in another PR